rpc/jsonrpc_server.go

```diff
@@ -10,12 +10,29 @@ import (
 
 	"github.com/ava-labs/avalanchego/ids"
 
+	"github.com/ava-labs/hypersdk/api"
 	"github.com/ava-labs/hypersdk/chain"
 	"github.com/ava-labs/hypersdk/codec"
 	"github.com/ava-labs/hypersdk/consts"
 	"github.com/ava-labs/hypersdk/fees"
 )
 
+var _ api.HandlerFactory[VM] = (*JSONRPCServerFactory)(nil)
+
+type JSONRPCServerFactory struct{}
```

Comment:
Thoughts on using this pattern here: https://github.com/ava-labs/avalanchego/pull/2717#discussion_r1486611976 ?

Reply:
Seems weird to me since either you hide the concrete type and the caller loses type information or you expose the type and the caller has two ways of doing the same thing. I understand the argument for the aesthetic look of it though... but personally not a huge fan of the pattern